### PR TITLE
feat: remove mask computations from forward

### PR DESCRIPTION
**Pull request by eduardocarvp**
**IMPORTANT: Please do not create a Pull Request without creating an issue first.**

*Any change needs to be discussed before proceeding. Failure to do so may result in the rejection of the pull request.*

**What kind of change does this PR introduce?**

**Does this PR introduce a breaking change?**

**What needs to be documented once your changes are merged?**

**Closing issues**

Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if such).

### CODE REVIEW

File: pytorch_tabnet/tab_network.py [hunk 1]
```diff
@@ -125,6 +125,29 @@ def __init__(self, input_dim, output_dim,
     def forward(self, x):
         res = 0
 
+        prior = torch.ones(x.shape).to(x.device)
+        M_loss = 0
+        att = self.initial_splitter(x)[:, self.n_d:]
+
+        for step in range(self.n_steps):
+            M = self.att_transformers[step](prior, att)
+            M_loss += torch.mean(torch.sum(torch.mul(M, torch.log(M+self.epsilon)),
```
[the remark this answers is not in the document]
Comment (eduardocarvp) [reply pytorch_tabnet/tab_network.py:134]: The whole thing does not seem to be much faster after all... The time for each epoch compared to develop is almost the same on my cpu tests

[the remark this answers is not in the document]
Comment (Optimox) [reply pytorch_tabnet/tab_network.py:134]: yeah and what happens with GPU?

File: pytorch_tabnet/tab_network.py [hunk 2]
```diff
@@ -232,6 +253,11 @@ def forward(self, x):
         x = self.initial_bn(x)
         return self.tabnet(x)
 
+    def forward_masks(self, x):
+        x = self.embedder(x)
```
[the remark this answers is not in the document]
Comment (eduardocarvp) [reply pytorch_tabnet/tab_network.py:259]: I moved `self.initial_bn(x)` to the parent class (had to move initial_bn's instantiation as well)

[the remark this answers is not in the document]
Comment (Optimox) [reply pytorch_tabnet/tab_network.py:259]: I got this error 
```---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
<ipython-input-10-84fdfa555c78> in <module>
      3     X_valid=X_valid, y_valid=y_valid,
      4     max_epochs=1, patience=100,
----> 5     batch_size=16384, virtual_batch_size=256, num_workers=0
      6 ) 

/work/pytorch_tabnet/tab_model.py in fit(self, X_train, y_train, X_valid, y_valid, loss_fn, weights, max_epochs, patience, batch_size, virtual_batch_size, num_workers, drop_last)
    209 
    210         # compute feature importance once the best model is defined
--> 211         self._compute_feature_importances(train_dataloader)
    212 
    213     def fit_epoch(self, train_dataloader, valid_dataloader):

/work/pytorch_tabnet/tab_model.py in _compute_feature_importances(self, loader)
    356         feature_importances_ = np.zeros((self.network.post_embed_dim))
    357         for data, targets in loader:
--> 358             M_explain, masks = self.network.forward_masks(data)
    359             feature_importances_ += M_explain.sum(dim=0).cpu().detach().numpy()
    360 

/work/pytorch_tabnet/tab_network.py in forward_masks(self, x)
    258     def forward_masks(self, x):
    259         x = self.embedder(x)
--> 260         return self.tabnet.forward_masks(x)
    261 
    262 

/work/pytorch_tabnet/tab_network.py in forward_masks(self, x)
    148 
    149     def forward_masks(self, x):
--> 150         x = self.initial_bn(x)
    151 
    152         prior = torch.ones(x.shape).to(x.device)

/work/.cache/poetry/pytorch-tabnet-DJpFP61h-py3.7/lib/python3.7/site-packages/torch/nn/modules/module.py in __call__(self, *input, **kwargs)
    530             result = self._slow_forward(*input, **kwargs)
    531         else:
--> 532             result = self.forward(*input, **kwargs)
    533         for hook in self._forward_hooks.values():
    534             hook_result = hook(self, input, result)

/work/.cache/poetry/pytorch-tabnet-DJpFP61h-py3.7/lib/python3.7/site-packages/torch/nn/modules/batchnorm.py in forward(self, input)
    105             input, self.running_mean, self.running_var, self.weight, self.bias,
    106             self.training or not self.track_running_stats,
--> 107             exponential_average_factor, self.eps)
    108 
    109 

/work/.cache/poetry/pytorch-tabnet-DJpFP61h-py3.7/lib/python3.7/site-packages/torch/nn/functional.py in batch_norm(input, running_mean, running_var, weight, bias, training, momentum, eps)
   1668     return torch.batch_norm(
   1669         input, weight, bias, running_mean, running_var,
-> 1670         training, momentum, eps, torch.backends.cudnn.enabled
   1671     )
   1672 

RuntimeError: "batch_norm" not implemented for 'Long' ```


If you lanch forest example with just one epoch you should be able to see the same error.

[the remark this answers is not in the document]
Comment (Optimox) [reply pytorch_tabnet/tab_network.py:259]: ok now it works!

**Pull request merged**